paasta_tools/secret_tools.py

```diff
@@ -0,0 +1,45 @@
+# Copyright 2015-2017 Yelp Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import os
+import re
+
+from paasta_tools.utils import SystemPaastaConfig
+
+SECRET_REGEX = "^SECRET\([A-Za-z0-9_-]*\)$"
+
+
+def is_secret_ref(val: str) -> bool:
+    pattern = re.compile(SECRET_REGEX)
+    return True if pattern.match(val) else False
+
+
+def get_hmac_for_secret(
+        val: str,
+        service: str,
+        soa_dir: str,
+        system_paasta_config: SystemPaastaConfig,
+) -> str:
+    secret_name = _get_secret_name_from_ref(val)
+    try:
+        secret_path = os.path.join(soa_dir, service, "secrets", "{}.json".format(secret_name))
+        with open(secret_path, 'r') as json_secret_file:
+            secret_file = json.load(json_secret_file)
+            return secret_file['environments'][system_paasta_config.get_vault_environment()]['signature']
+    except IOError as e:
+        print("Failed to load json secret at {}".format(secret_path))
+        return None
```

Comment:
If you're returning `None` the type of the function needs to be `Optional[str]`

Reply:
Interesting, I also don't have a test that exercises this, maybe it would through a type error if I did...

Reply:
I think you'll only get an error on `tox -e mypy`, but only once you add this file to that section of `tox.ini`

Reply:
Huh, weird I added it to the tox.ini and some tests but it still didn't complain. Although it also doesn't complain with Optional[str]. Anyway, I'm sure it should be optional!